packages/back-end/src/integrations/Snowflake.ts

```diff
@@ -31,4 +40,60 @@ export default class Snowflake extends SqlIntegration {
   ensureFloat(col: string): string {
     return `CAST(${col} AS DOUBLE)`;
   }
+  async getInformationSchema(): Promise<InformationSchema[]> {
+    const database = this.params.database;
+
+    if (!database) {
+      throw new Error(
+        "No database provided. In order to get the information schema, you must provide a database."
+      );
+    }
+
+    const queryString = `SELECT
+        table_name,
+        table_catalog,
+        table_schema,
+        count(column_name) as column_count
+    FROM
+        ${database}.INFORMATION_SCHEMA.COLUMNS
```

Comment:
@jdorn It feels like we might need to do some escaping here - I was initially thinking we could use Snowflake's `$$` to wrap `${database}.INFORMATION_SCHEMA.COLUMNS` but I haven't been able to get it to work within the `FROM` clause, only below in the `WHERE IN` clauses.

Am I correct in thinking we need to escape this or no?

Reply:
You're worried that their `database` has an `'` in it, or the like? I would guess that's not possible, and that the escaping is for strings, and not for resource names.

I'd think we don't need to escape the resource name.

Reply:
Yeah, BigQuery is the only one where I see this quoted since it's common to have hyphens and other unusual characters in project/dataset names.